src/node/resolver.ts

```diff
@@ -165,27 +174,21 @@ export function createResolver(
     if (!resolved) {
       resolved = defaultRequestToFile(publicPath, root)
     }
-    const ext = resolveExt(resolved)
-    const result = {
-      filePath: ext ? resolved + ext : resolved,
-      ext: ext || path.extname(resolved)
```

Comment:
Previous code can't guarantee ext is the **actual missing postfix of publicPath**, because `defaultRequestToFile` would add `/index.js` to `resolved ` but ext still return `.js`.